core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1876,17 +1822,6 @@ private[spark] object Utils extends Logging {
     obj.getClass.getSimpleName.replace("$", "")
   }
 
-  /** Return an option that translates JNothing to None */
-  def jsonOption(json: JValue): Option[JValue] = {
```

Comment:
This is only used in `JSONProtocol` for now, but this is also a useful JSON method that could apply to some other JSON operations in the future.

Reply:
We should not be using json4s for new things, so it's better to have these hidden away.

Reply:
What is the reason we should not use json4s for new things?

Reply:
It encourages writing manual serialization and deserialization code, which is error prone.